src/libstd/os/linux/fs.rs

```diff
@@ -24,9 +24,27 @@ pub trait MetadataExt {
     /// Gain a reference to the underlying `stat` structure which contains
     /// the raw information returned by the OS.
     ///
-    /// The contents of the returned `stat` are **not** consistent across
+    /// The contents of the returned [`stat`] are **not** consistent across
     /// Unix platforms. The `os::unix::fs::MetadataExt` trait contains the
     /// cross-Unix abstractions contained within the raw stat.
+    ///
+    /// [`stat`]: ../../../../std/os/linux/raw/struct.stat.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #[cfg(target_os = "linux")] {
```

Comment:
This `#[cfg(target_os = "linux")]` block could be removed, `#[doc(cfg)]` should not run the doc tests outside of Linux.

Reply:
The point is that, if users want to take a look at the full code (by clicking on "run"), they'll see this conditional compilation.

Reply:
How many people click on "Run" though? I'm thinking of how many people copy out the examples in `File` and run into "? in main" errors.